### DISCUSSION TITLE
Data loss in indexDB

### DISCUSSION THREAD

**Discussion by wy2010344** [ujjwalguptaofficial/JsStore#286] [Q&A]
Is there any bug here causing the database to be rebuilt?
My data in indexDB is all lost.

I built an app using vite+pwa.
Always run offline.
I opened it yesterday and found that the data was missing.
Of course, I feel that the problem may also be in pwa.
Recently it opened first to enter 404, and then enter the normal interface.
When I opened it yesterday, I felt that the home page routing had changed.When I changed it to the correct route, the data was lost.
android+chrome 103.

```
  "devDependencies": {
    "typescript": "^4.6.4",
    "vite": "^3.0.0",
    "vite-plugin-pwa": "^0.12.3"
  },
  "dependencies": {
    "better-react": "^1.0.5",
    "better-react-dom": "^1.0.5",
    "better-react-helper": "^1.0.5",
    "dayjs": "^1.11.4",
    "idb": "^7.0.2",
    "jsstore": "^4.4.1",
    "socket.io-client": "^4.5.1",
    "stylis-creater": "^1.0.5"
  }
```
Do you know any possibility?

**Comment by ujjwalguptaofficial** [comment]
yeah browser can remove data, when storage is not available. Check this article - https://developer.mozilla.org/en-US/docs/Web/API/IndexedDB_API/Browser_storage_limits_and_eviction_criteria#lru_policy
